app/models/ebean/ReportExecution.java

```diff
@@ -76,7 +74,7 @@
     @Nullable
     @DbJsonB
     @Column(name = "error")
-    private Map<String, String> error;
+    private Error error;
```

Comment:
Is this data-type change... legal? Don't we need a migration, at least?
(I see your new Error classes below, but I don't think I understand how this all fits together. Help?)

Reply:
This isn't referring to:

https://docs.oracle.com/javase/7/docs/api/java/lang/Error.html

Is it?

I would also like to know how this is backwards compatible and what motivated the change.

Reply:
This is backwards compatible and I included a test for it. I changed it because while revisiting the existing structure I wanted something more structured than `Map<String, String>`

Reply:
I *believe* the original point of this was to be that unstructured so that it could store more than just an exception. For example, we could store progress, messages from dependencies, etc..

Reply:
I figure that with this change you'd still be able to throw in new fields in a new structure that would be identified by the `JsonTypeInfo`. Yes it would not be as flexible for throwing in these other fields but we could still handle those other cases with additional metadata by adding a new class to hold them.

I suppose it's all trade-offs. If there's no obvious benefit here then I can just revert this.